client/src/components/BulkCreateForm/BulkCreateForm.js

```diff
@@ -0,0 +1,226 @@
+
+    /**
+     * Resets the state of the Form
+     * This means resetting the validation state for all inputs to true and clearing all error messages
+     * @param resetValue - If true, also resets the values of the inputs
+     */
+    resetForm(resetValue) {
+        const { fileUpload } = this.state;
+        let { fileUploadInputKey } = this.state;
+
+        fileUpload.isValid = true;
+        fileUpload.errorMessage = '';
+
+        if (resetValue) {
+            fileUpload.file = {};
+            fileUpload.message = 'No file chosen';
+            fileUploadInputKey++;
+        }
+
+        this.setState({fileUpload, fileUploadInputKey});
+    }
+
+    handleCancelButtonClick() {
+        const { onCancelButtonClick } = this.props;
+
+        onCancelButtonClick();
+    }
+
+    /**
+     * Validates the Form
+     * If any of the inputs are invalid, the entire Form is considered invalid
+     * @returns {boolean} - True if the Form is valid, False otherwise
+     */
+    validateForm() {
+        const { fileUpload } = this.state;
+        let formIsValid = true;
+
+        /* File Upload Validation */
+        /* A File upload is required */
+        if (fileUpload.file.size === undefined) {
+            fileUpload.isValid = false;
+            fileUpload.errorMessage = 'A file is required';
+            formIsValid = false;
+        }
+
+        if (!formIsValid) {
+            this.setState({fileUpload});
+        }
+
+        return formIsValid;
+    }
+
+    /**
+     * Build the form object to send to the onSubmit handler
+     */
+    submitForm() {
+        const { fileUpload } = this.state;
+        const { onSubmit } = this.props;
+
+        let form = {
+            file: fileUpload.file
+        };
+
+        onSubmit(form);
+    }
+
+    handleFeedbackModalClose() {
+        this.resetForm(true);
+    }
+
+    render() {
+        const { bulkCreateResult } = this.props;
+        const { fileUpload, showingMoreInformation, fileUploadInputKey } = this.state;
+
+        const moreInformationLink = (
+            <div className='more-information-link'
+                 onClick={() => this.handleMoreInformationClick()}>
+                Show More Information
+            </div>
+        );
+
+        const hideMoreInformationLink = (
+            <div className='more-information-link hide-link'
+                 onClick={() => this.handleMoreInformationClick()}>
+                Hide More Information
+            </div>
+        );
+
+        return (
+            <div className='bulk-create-form-container'>
+                <div className='h5'>
+                    Bulk Create Monuments and Memorials
+                </div>
+
+                <Form onSubmit={(event) => this.handleSubmit(event)}>
+                    <Form.Group className='file-upload-form-group'>
+                        <Form.Label>Upload a CSV File:</Form.Label>
+                        <label htmlFor='file-upload-input' className='file-upload-input-label'>
```

Comment:
I overlaid a `<label>` on this HTML file input because you can't style the input.